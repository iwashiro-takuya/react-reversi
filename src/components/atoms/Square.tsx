import React from 'react';
import styled from 'styled-components';
import Const from '../../const';

const { Color } = Const;

const Square: React.FC = ({ children }) => {
  return <Wrapper>{children}</Wrapper>;
};

const Wrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 80px;
  height: 80px;
  background-color: ${Color.BG_GREEN};
  border: 1px solid ${Color.BD_BLACK};
`;

export default Square;
