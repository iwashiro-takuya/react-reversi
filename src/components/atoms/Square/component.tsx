import React from 'react';
import styled from 'styled-components';
import Const from 'src/const';

const { Size, Color } = Const;

export interface SquareProp {
  dataCy: string;
}

const Square: React.FC<SquareProp> = ({ children, dataCy }) => {
  return <Wrapper data-cy={dataCy}>{children}</Wrapper>;
};

const Wrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${Size.SQUARE_SIZE}px;
  height: ${Size.SQUARE_SIZE}px;
  background-color: ${Color.BG_GREEN};
  border: 1px solid ${Color.BD_BLACK};
`;

export default Square;
