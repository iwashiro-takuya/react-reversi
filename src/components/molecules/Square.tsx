import React from 'react';
import styled from 'styled-components';
import Piece from '../atoms/Piece';
import Const from '../../const';

const { Color } = Const;

interface Square {
  key: number;
  piece: 'none' | 'block';
}

const Square: React.FC<Square> = ({ piece }) => {
  const color: string = Color.PC_INVISIBLE;
  // switch (value) {
  //   case 1:
  //     color = Color.PC_WHITE;
  //     break;
  //   case -1:
  //     color = Color.PC_BLACK;
  //     break;
  //   default:
  //     color =
  // }

  return (
    <Wrapper>
      <Piece display={piece} color={color} />
    </Wrapper>
  );
};

const Wrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 80px;
  height: 80px;
  background-color: ${Color.BG_GREEN};
  border: 1px solid ${Color.BD_BLACK};
`;

export default Square;
