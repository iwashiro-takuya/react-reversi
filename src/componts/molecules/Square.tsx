import React from 'react';
import styled from 'styled-components';
import Const from '../../const';

import Piece from '../atoms/Piece';

const { Color } = Const;

export default () => {
  return (
    <Wrapper>
      <Piece color=""></Piece>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  align-items: center;
  background-color: ${Color.BG_GREEN};
  border: 1px solid ${Color.BD_BLACK};
  box-sizing: border-box;
  display: flex;
  height: 80px;
  justify-content: center;
  width: 80px;
`;
