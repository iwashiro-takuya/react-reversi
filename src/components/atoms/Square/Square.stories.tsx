import React, { ComponentProps } from 'react';
import { Story } from '@storybook/react/types-6-0';
import Square from './component';

export default {
  title: 'Square',
  component: Square,
};

const Template: Story<ComponentProps<typeof Square>> = (args) => (
  // eslint-disable-next-line react/jsx-props-no-spreading
  <Square {...args} />
);

export const FirstStory = Template.bind({});

FirstStory.args = {
  dataCy: 'dataCy',
};
