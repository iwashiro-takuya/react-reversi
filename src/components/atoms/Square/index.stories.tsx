import React from 'react';
import { Story } from '@storybook/react/types-6-0';
import Square, { SquareProps } from './component';

export default {
  title: 'Atoms/Square',
  component: Square,
};

const Template: Story<SquareProps> = () => <Square dataCy="dataCy" />;

export const normal = Template.bind({});
