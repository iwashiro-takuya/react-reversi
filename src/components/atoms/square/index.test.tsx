import React from 'react';
import renderer from 'react-test-renderer';
import Square from '.';

describe('Square unit test', () => {
  it('snapshot', () => {
    const component = renderer.create(<Square dataCy="square-1" />);
    expect(component).toMatchSnapshot();
  });
});
