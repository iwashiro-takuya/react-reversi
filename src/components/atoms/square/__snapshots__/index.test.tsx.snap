// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square unit test snapshot 1`] = `
<div
  className="sc-bdnxRM cmyxMg"
  data-cy="square-1"
/>
`;
